amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -182,6 +186,19 @@ private void createLeasesForChildShardsIfNotExist()
         }
     }
 
+    private void updateLeasesForChildShards()
+            throws DependencyException, InvalidStateException, ProvisionedThroughputException {
+        Lease currentLease = leaseCoordinator.getCurrentlyHeldLease(shardInfoIdProvider.apply(shardInfo));
+        Set<String> childShardIds = new HashSet<>();
+        for (ChildShard childShard : childShards) {
```

Comment:
I think we can also make this something like `childShards.stream().map(ChildShard::shardId).collect(Collectors.toSet())`